frontends/mit-open/src/pages/DashboardPage/Dashboard.test.tsx

```diff
@@ -74,20 +220,89 @@ describe("DashboardPage", () => {
+      const desktopTab = screen.getByTestId(`desktop-tab-${key}`)
+      const mobileTab = screen.getByTestId(`mobile-tab-${key}`)
+      expect(desktopTab).toBeInTheDocument()
+      expect(mobileTab).toBeInTheDocument()
+      expect(desktopTab).toHaveAttribute("href", `/#${key}`)
+      expect(mobileTab).toHaveAttribute("href", `/#${key}`)
     })
+  })
 
-    renderTestApp({
-      url: "/dashboard",
+  test("Renders the expected carousels on the home page", async () => {
+    const {
+      profile,
+      topPicks,
+      topicsCourses,
+      certificationCourses,
+      courses,
+      resources,
+    } = setupAPIs()
+    const all = [...courses, ...resources.results]
+    renderWithProviders(<DashboardPage />)
+
+    const topPicksTitle = await screen.findByText("Top picks for you")
+    expect(topPicksTitle).toBeInTheDocument()
+    const topPicksCarousel = await screen.findByTestId("top-picks-carousel")
+    topPicks.results.forEach(async (course) => {
+      const courseTitle = await within(topPicksCarousel).findByText(
+        course.title,
+      )
+      expect(courseTitle).toBeInTheDocument()
     })
-    Object.keys(DashboardTabLabels).forEach(async (key) => {
-      const desktopTab = await screen.findByTestId(`desktop-tab-${key}`)
-      const mobileTab = await screen.findByTestId(`mobile-tab-${key}`)
-      expect(desktopTab).toBeInTheDocument()
-      expect(mobileTab).toBeInTheDocument()
-      expect(desktopTab).toHaveAttribute("href", `/dashboard#${key}`)
-      expect(mobileTab).toHaveAttribute("href", `/dashboard#${key}`)
+
+    profile.preference_search_filters.topic?.forEach(async (topic) => {
+      const topicTitle = await screen.findByText(`Popular courses in ${topic}`)
+      expect(topicTitle).toBeInTheDocument()
+      const topicCarousel = await screen.findByTestId(`topic-carousel-${topic}`)
+      topicsCourses
+        .filter((course) => course.topics?.[0].name === topic)
+        .forEach(async (course) => {
+          const courseTitle = await within(topicCarousel).findByText(
+            course.title,
+          )
+          expect(courseTitle).toBeInTheDocument()
+        })
+    })
+
+    const certificationText = `Courses ${profile.preference_search_filters.certification ? "with" : "without"} Certificates`
+    const certificationTitle = await screen.findByText(certificationText)
+    expect(certificationTitle).toBeInTheDocument()
+    const certificationCarousel = await screen.findByTestId(
+      "certification-carousel",
+    )
+    certificationCourses.results.forEach(async (course) => {
+      const courseTitle = await within(certificationCarousel).findByText(
+        course.title,
+      )
+      expect(courseTitle).toBeInTheDocument()
+    })
+
+    const newTitle = await screen.findByText("New")
+    expect(newTitle).toBeInTheDocument()
+    const newCarousel = await screen.findByTestId(
+      "new-learning-resources-carousel",
+    )
+    all.forEach(async (learningResource) => {
```

Comment:
I suggest changing this to 
```
await Promise.all(all.map(...))
```
I'm not entirely sure it's guaranteed that these promises resolve before the test finishes.


That said, I'm not sure it's worth making very detailed assertions about the carousel like this. The resource carousel has its own tests. I generally do support the philosophy of "testing what the user should see", but for something we (1) test separately and (2) reuse so much, it seems appropriate to me to do

```ts
jest.mocked("@/page-components/ResourceCarousel/ResourceCarousel", () => {
  const actual = jest.requireActual(
    "@/page-components/ResourceCarousel/ResourceCarousel",
  )
  return {
    ...actual,
    __esModule: true,
    default: jest.fn(actual.default),
  }
})
const spyCarousel = jest.mocked(ResourceCarousel)
```
and asserting that the carousel has been passed appropriate configs based on user profile.

**OR, even simpler** Mock the carousel, don't just spy on it:
```
jest.mocked("@/page-components/ResourceCarousel/ResourceCarousel", () => {
  const actual = jest.requireActual(
    "@/page-components/ResourceCarousel/ResourceCarousel",
  )
  return {
    ...actual,
    __esModule: true,
    default: jest.fn(() => (<div />)), // I think you need to return a JSX element or react gets mad. I forget.
  }
})
```

The carousel is something we use so much, I think this is probably the way we should handle testing it on individual pages.